binary/type.proto

```diff
@@ -154,4 +165,66 @@ message Type {
         int32 organization = 1;
         string name = 2;
     }
+
+    message IntegerParameter {
+        oneof integer_type {
+            int32 literal = 1;
+            String parameter_name = 2;
+        }
+    }
 }
+
+
+message TypeExpression {
+
+    oneof expression_type {
+        int32 integer_literal = 1;
+        UnaryOp unary_op = 2;
+        BinaryOp binary_op = 3;
+        Type.Parameter parameter = 4;
+        IfElse if_else = 5;
+        NotBindable not_bindable = 6;
+        Type type = 7;
+    }
+
+    message IfElse {
+        TypeExpression if_condition = 1;
+        TypeExpression if_return = 2;
```

Comment:
Is this the type of "then" expressions? Why is it not called then_return? Shouldn't the type of "then" and "else" expressions be the same? If so, why do we need both if_return and else_return?